spec/tapioca/cli/gem_spec.rb

```diff
@@ -1466,6 +1466,54 @@ class User; end
           assert_empty_stderr(result)
           assert_success_status(result)
         end
+
+        it "generates `ObjectSpace::WeakMap` as a generic type with Sorbet < 0.5.10587" do
+          foo = mock_gem("foo", "0.0.1") do
+            write("lib/foo.rb", <<~RB)
+              Foo = ObjectSpace::WeakMap.new
+            RB
+          end
+
+          @project.gemfile(@project.tapioca_gemfile)
+          @project.require_mock_gem(foo)
+          @project.require_real_gem("sorbet-static", "0.5.10585")
+          @project.bundle_install
+          @project.tapioca("gem")
```

Comment:
Test failure is caused by `sorbet-static-and-runtime` version not being satisfied during tapioca execution. Mock project depends on tapioca which brings in the Sorbet version that project uses. In the failing CI builds Sorbet version is higher than `0.5.10585` and as a result `bundle install` fails since it didn't bump it.

We can run a `bundle update` after requiring `sorbet-static-and-runtime` instead:
```
          @project.gemfile(@project.tapioca_gemfile)
          @project.require_mock_gem(foo)
          @project.bundle_install
          @project.require_real_gem("sorbet-static-and-runtime", "0.5.10585")
          @project.bundle_update("sorbet-static-and-runtime")
          @project.tapioca("gem")
```

`bundle_update` can be implemented similar to [bundle_install](https://github.com/Shopify/tapioca/blob/main/spec/helpers/mock_project.rb#L89-L109). Maybe it's okay to couple this with `require_real_gem` and trigger it within that method.

Reply:
Hmm, however the test setup should not work that way. The dependency on Tapioca is now via the `.gemspec` file, which should not limit the `sorbet-static-and-runtime` version selection. Not sure what's going wrong here.

Reply:
Ok, it is about the `mock_project` bringing in the sorbet dependency automatically. The way around it is to use the `mock_project(sorbet_dependency: false)` flag, like it was here: https://github.com/Shopify/tapioca/pull/1260/files#diff-bee8bd10954cc2dcd94f010d4bc1b3092af1883d0494cfda0c1b653718550b11L1875

Reply:
I pushed a commit to fix that

Reply:
Good find, this is a much better solution.